### PR TITLE
feat(build): updating to rust compiler 1.72.0 and latest tools

### PR DESCRIPTION
**Pull request by dsavitskiy**
Updated the following tool version:
* Rust toolchain Nix overlay: 2023-08-25
* Stable rustc 1.72.0 (5680fa18f 2023-08-23)
* cargo 1.74.0-nightly
* rustfmt 1.6.0-nightly
* clippy 0.1.73

Code fixed to meet new clippy rules.

### CODE REVIEW

**Comment by dsavitskiy** [comment]
bors try

**Comment by dsavitskiy** [comment]
bors try

**Comment by dsavitskiy** [comment]
bors try

**Comment by dsavitskiy** [comment]
bors try

File: io-engine/src/rebuild/rebuild_descriptor.rs [hunk 1]
```diff
@@ -44,6 +44,9 @@ pub(super) struct RebuildDescriptor {
     pub(super) rebuild_map: Arc<parking_lot::Mutex<Option<RebuildMap>>>,
 }
 
+unsafe impl Sync for RebuildDescriptor {}
```
Comment (tiagolobocastro) [inline io-engine/src/rebuild/rebuild_descriptor.rs]: Why are these needed?

**Review by tiagolobocastro — APPROVED** [review]

File: io-engine/src/rebuild/rebuild_descriptor.rs (hunk 1 above)
Comment (dsavitskiy) [reply io-engine/src/rebuild/rebuild_descriptor.rs]: Ah, let me check, probably it is not needed. At some point, compiler complained about but I changed implementation since then.

**Review by mtzaurus — APPROVED** [review]

File: io-engine/src/rebuild/rebuild_descriptor.rs (hunk 1 above)
Comment (dsavitskiy) [reply io-engine/src/rebuild/rebuild_descriptor.rs]: Fixed.

**Comment by dsavitskiy** [comment]
bors merge
